### PR TITLE
[processor/resourcedetection] Add host.ip addresses detection

### LINKED ISSUE #24765: [processor/resourcedetection] Add `host.ip` to system detector

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:** 

Adds detection for `host.ip`.

Relates to open-telemetry/semantic-conventions/pull/203

Fixes #24765

### CODE REVIEW

File: .chloggen/mx-psi_host.ip-ref-impl.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Use this changelog template to create an entry for release notes.
+# If your change doesn't affect end users, such as a test fix or a tooling change,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhacnenment
```
Comment (mx-psi) [inline .chloggen/mx-psi_host.ip-ref-impl.yaml:6]:
```suggestion
change_type: enhancement
```

**Comment by mx-psi** [comment]
cc @ChrsMark

File: internal/metadataproviders/system/metadata.go [hunk 2]
```diff
@@ -63,6 +63,9 @@ type Provider interface {
 
 	// HostArch returns the host architecture
 	HostArch() (string, error)
+
+	// HostIPv4Addresses returns the host's IPv4 interfaces
```
Comment (ChrsMark) [inline internal/metadataproviders/system/metadata.go:67]: I guess this docstring needs to be updated?

File: internal/metadataproviders/system/metadata.go [hunk 3]
```diff
@@ -159,3 +162,37 @@ func (p systemMetadataProvider) OSDescription(ctx context.Context) (string, erro
...
+		for _, addr := range addrs {
+			ip, _, parseErr := net.ParseCIDR(addr.String())
+			if parseErr != nil {
+				return nil, fmt.Errorf("failed to parse address %q from interface %v: %w", addr, iface, parseErr)
+			}
+
+			if ip.IsLoopback() {
+				// skip loopback IPs
+				continue
+			}
```
Comment (frzifus) [inline internal/metadataproviders/system/metadata.go:191]: Is there a chance to have a loopback IP on an non loopback interface?

Comment (mx-psi) [reply internal/metadataproviders/system/metadata.go:191]: I guess that is possible, but I couldn't find a way to test this. `IsLoopback` will test if the IP belongs to the `127.0.0.0/8` range (or the equivalent in IPv6), I think that is the best we can do.

File: internal/metadataproviders/system/metadata.go (hunk 2 above)
Comment (mx-psi) [reply internal/metadataproviders/system/metadata.go]: Indeed, fixed!

**Review by ChrsMark — APPROVED** [review]
lgtm. Code owners should also have a look :)

**Comment by mx-psi** [comment]
@frzifus can you take another look?

**Review by frzifus — APPROVED** [review]
LGTM :)

File: processor/resourcedetectionprocessor/internal/system/system.go [hunk 4]
```diff
@@ -71,6 +71,15 @@ func (d *Detector) Detect(ctx context.Context) (resource pcommon.Resource, schem
...
 	}
 
+	var hostIPAttribute []any
+	hostIPs, errIPs := d.provider.HostIPs()
+	if errIPs != nil {
+		return pcommon.NewResource(), "", fmt.Errorf("failed getting host IP addresses: %w", errIPs)
+	}
+	for _, ip := range hostIPs {
+		hostIPAttribute = append(hostIPAttribute, ip.String())
+	}
```
Comment (dmitryax) [inline processor/resourcedetectionprocessor/internal/system/system.go:81]: I'd be good to wrap it in `d.cfg.ResourceAttributes.HostIP.Enabled` so errors are not returned if the attribute is disabled

Comment (mx-psi) [reply processor/resourcedetectionprocessor/internal/system/system.go]: Makes sense, done

**Review by atoulme — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

**Pull request merged**